htdocs/admin/remotestore/class/externalModules.class.php

```diff
@@ -0,0 +1,526 @@
+		return $ret;
+	}
+
+	// phpcs:disable PEAR.NamingConventions.ValidFunctionName.ScopeNotCamelCaps
+	/**
+	 * get previous link
+	 *
+	 * @param   string    $text     symbol previous
+	 * @return  string              html previous link
+	 */
+	public function get_previous_link($text = '<<')
+	{
+		// phpcs:enable
+		return '<a href="'.$this->get_previous_url().'" class="button">'.dol_escape_htmltag($text).'</a>';
+	}
+
+	// phpcs:disable PEAR.NamingConventions.ValidFunctionName.ScopeNotCamelCaps
+	/**
+	 * get next link
+	 *
+	 * @param   string    $text     symbol next
+	 * @return  string              html next link
+	 */
+	public function get_next_link($text = '>>')
+	{
+		// phpcs:enable
+		return '<a href="'.$this->get_next_url().'" class="button">'.dol_escape_htmltag($text).'</a>';
+	}
+
+	// phpcs:disable PEAR.NamingConventions.ValidFunctionName.ScopeNotCamelCaps
+	/**
+	 * get previous url
+	 *
+	 * @return string    previous url
+	 */
+	public function get_previous_url()
+	{
+		// phpcs:enable
+		$param_array = array();
+		if ($this->no_page > 1) {
+			$sub = 1;
+		} else {
+			$sub = 0;
+		}
+		$param_array['no_page'] = $this->no_page - $sub;
+		if ($this->categorie != 0) {
+			$param_array['categorie'] = $this->categorie;
+		}
+		$param = http_build_query($param_array);
+		return $this->url."&".$param;
+	}
+
+	// phpcs:disable PEAR.NamingConventions.ValidFunctionName.ScopeNotCamelCaps
+	/**
+	 * get next url
+	 *
+	 * @return string    next url
+	 */
+	public function get_next_url()
+	{
+		// phpcs:enable
+		$param_array = array();
+		if ($this->products !== null && count($this->products) < $this->per_page) {
+			$add = 0;
+		} else {
+			$add = 1;
+		}
+		$param_array['no_page'] = $this->no_page + $add;
+		if ($this->categorie != 0) {
+			$param_array['categorie'] = $this->categorie;
+		}
+		$param = http_build_query($param_array);
+		return $this->url."&".$param;
+	}
+
+	/**
+	 * Check the status code of the request
+	 *
+	 * @param array{status_code:int,response:?string,header:string} $request Response elements of CURL request
+	 *
+	 * @return string|null
+	 */
+	protected function checkStatusCode($request)
+	{
+		// Define error messages
+		$error_messages = [
+			204 => 'No content',
+			400 => 'Bad Request',
+			401 => 'Unauthorized',
+			404 => 'Not Found',
+			405 => 'Method Not Allowed',
+			500 => 'Internal Server Error',
+		];
+
+		// If status code is 200 or 201, return an empty string
+		if ($request['status_code'] === 200 || $request['status_code'] === 201) {
+			return '';
+		}
+
+		// Get the predefined error message or use a default one
+		$error_message = $error_messages[$request['status_code']] ?? 'Unexpected HTTP status: ' . $request['status_code'];
+
+		// Append error details if available
+		if (!empty($request['response']) && isset($request['response']['errors']) && is_array($request['response']['errors'])) {
```

Comment:
It is indicated that 'response' is a string.  So to fix that you need to suggest it is `response:?array{errors:string[]}` or `response:null|string|array{errors:string[]}` if it can also be a string.